cpp/foxglove/src/channel.cpp

```diff
@@ -14,9 +15,13 @@ Channel::Channel(
     cSchema.data = reinterpret_cast<const uint8_t*>(schema->data);
     cSchema.data_len = schema->dataLen;
   }
-  _impl.reset(
-    foxglove_channel_create(topic.data(), messageEncoding.data(), schema ? &cSchema : nullptr)
-  );
+  foxglove_error cError = {};
+  _impl.reset(foxglove_channel_create(
+    topic.data(), messageEncoding.data(), schema ? &cSchema : nullptr, &cError
+  ));
+  if (_impl.get() == nullptr) {
```

Comment:
nit: [can be shortened](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool) (applies to a few other places too)
```suggestion
  if (!_impl) {
```